src/format/kdbx4/dump.rs

```diff
@@ -121,15 +154,19 @@ impl KDBX4Header {
 }
 
 impl KDBX4InnerHeader {
-    fn dump(&self, writer: &mut dyn Write) -> Result<(), DatabaseSaveError> {
+    fn dump(
+        &self,
+        binaries: &BinaryAttachments,
+        writer: &mut dyn Write,
+    ) -> Result<(), DatabaseSaveError> {
         writer.write(&[INNER_HEADER_RANDOM_STREAM_ID])?;
         writer.write_u32::<LittleEndian>(4)?;
         writer.write_u32::<LittleEndian>(self.inner_random_stream.dump())?;
 
         writer.write_u8(INNER_HEADER_RANDOM_STREAM_KEY)?;
         writer.write_with_len(&self.inner_random_stream_key)?;
 
-        for binary in &self.binaries {
+        for binary in &binaries.binaries {
```

Comment:
@sseemayer maybe the parameter could be called `attachments` instead? This would make the current referencing less awkward.

Reply:
This should be addressed with #120